### DISCUSSION TITLE
Pyosmium on windows 11

### DISCUSSION THREAD

**Discussion by Ebe66** [osmcode/pyosmium#248] [Q&A] [answered]
I just can't figure it out and I hope somebody can get me on the right track. I'm on windows 11 and Python 3.9.

I have been creating custom maps for the Wahoo line of bike computers since mid 2021. Processing osm.pbf files from Geofabrik to .map files with the Mapsforge mapwriter plugin. The only thing that eluded me was the processing of relations. For example for route=bicycle relations. I found a repository that will/should process these, but it uses pyosmium.  [https://github.com/darauble/Locus-Map-DIY](url) which has a parse-routes.py file which should get me on the right way.
I did pip install osmium and it shows 3.7.0 is successfully installed. However, when I try to run that Python file it returns a "AttributeError: module 'osmium' has no attribute 'FileProcessor'" error. 
Ok, so far not so good ;-) So I thought let's try an example from this repository, but it just results in other errors.
The pub_names.py example (and others) result in this error: AttributeError: module 'osmium' has no attribute 'FileProcessor'.
I'm sure it's just silly me, but what am I doing wrong?

**Comment by lonvia** [comment, marked as answer]
You have gotten something mixed up with your installation "FileProcessor" only exists in the current master version, not in released versions. You should only used the release with the Locus-Map_DIY repo. If you have cloned this repository, you might want to get rid of it.

**Reply by Ebe66** [reply]
Thanks for the response! But I'm a beginner at best, could you elaborate a little?
The "fileprocessor" call is what is being used in the Locus-Map_DIY repo, I did not come up with that myself. 
I did not clone that repo, I just downloaded the Python file and tried to run it, resulting in the error.
My next step was to try some of the pyosmium examples, but they also don't work for me.  
For example "python pub_names.py zuid-holland-latest.osm.pbf" results in the same error.
I did a "pip uninstall osmium" followed by a "pip install osmium", but no change in the result. Should I be doing something else? Remember I am on windows, which seems to be ehmm, not so common in the mapping world... unfortunately (for me).

**Reply by Ebe66** [reply]
Sorry, but I totally got things mixed up. Is it possible to delete this question?

**Reply by lonvia** [reply]
We usually don't delete questions. Don't worry if the question may seem to look a bit silly now. Everybody gets things mixed up from time to time and you won't be the last to run into this particular problem. Ideally, you answer yourself here and describe what you did wrong and how you fixed it. The next person to run into the same trouble will be grateful when they get the right pointers.

**Comment by Ebe66** [comment]
OK, well as I said, I am a beginner at best. Things went wrong for my starting with this line in the python program from the link in my first post: osmium.io.Reader(file_name) This one still doesn't work for me btw. 
I started searching and got to this repository. I have pyosmium 3.7.0 installed and tried some of the examples which also did not work and resulted in a "AttributeError: module 'osmium' has no attribute 'FileProcessor'" error. At first i didn't understand Ionvia's answer, but when I did I tried an example from a earlier version of this repository and that worked. After that I got the parts I needed from the Locus-Map_DIY repo working.
That is mostly, I still have a very strange issue but that is not for this repo I guess. The python program works on Ubuntu. It also works on windows but the program never ends on windows. The last line of code is executed but it never returns to the dos prompt...
